urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/parse/stm/ForEachParser.java

```diff
@@ -0,0 +1,72 @@
+package parse.stm;
+
+import common.NiftyException;
+
+import parse.*;
+import parse.exp.ExpressionParser;
+import parse.misc.*;
+import a.Type;
+import a.exp.Expression;
+import a.stm.*;
+
+public class ForEachParser extends Parser<Statement> {
+    public static final Parser<Statement> singleton = new ForEachParser();
+    private ForEachParser() {}
+
+    @Override
+    public Success<Statement> parse(String s, int p) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 48-70 into a separate method.